src/runtime/classmanager.cs

```diff
@@ -470,6 +470,17 @@ private static ClassInfo GetClassInfo(Type type)
 
                 ob = new MethodObject(type, name, mlist);
                 ci.members[name] = ob;
+                if (mlist.Any(OperatorMethod.IsOperatorMethod))
+                {
+                    string pyName = OperatorMethod.GetPyMethodName(name);
+                    string pyNameReverse = pyName.Insert(2, "r");
+                    MethodInfo[] forwardMethods, reverseMethods;
+                    OperatorMethod.FilterMethods(mlist, out forwardMethods, out reverseMethods);
+                    // Only methods where the left operand is the declaring type.
+                    ci.members[pyName] = new MethodObject(type, name, forwardMethods);
+                    // Only methods where only the right operand is the declaring type.
+                    ci.members[pyNameReverse] = new MethodObject(type, name, reverseMethods);
```

Comment:
What if there are no reverseMethods?

Reply:
Currently it will just create a method that's never used; I can fix that by adding a hasReverse field in the SlotDefinition struct.

Reply:
So if C# reverse operator isn't defined, so the Python reverse name will have an empty method list in MethodObject

So if a user calls `b.__radd__(a)` by writing `a + b` where there is no forward +() defined on `a` so Python resorts to __radd__, AND no reverse +() defined on `b`, what happens?

Another unrelated issue is that of doing unnecessary work, if the Python slot doesn't support reverse e.g. `__truediv__`

Reply:
It is also possible to have no "forward" methods.

If no reverse methods are defined, the corresponding slot should not be set. Same for forward.

> Python slot doesn't support reverse e.g. `__truediv__`

Not sure what you mean. There's `__rtruediv__`

Reply:
That's true, I've just put a check then to not set the slot method if the method list is empty